Libplanet.Net/Consensus/HeightVoteSet.cs

```diff
@@ -0,0 +1,284 @@
+                    if (!_peerCatchupRounds.ContainsKey(validatorKey))
+                    {
+                        _peerCatchupRounds[validatorKey] = new List<int>();
+                    }
+
+                    List<int> rounds = _peerCatchupRounds[validatorKey];
+                    if (rounds.Count < 2)
+                    {
+                        AddRound(vote.Round);
+                        voteSet = GetVoteSet(vote.Round, vote.Flag);
+                        rounds.Add(vote.Round);
+                        _peerCatchupRounds[validatorKey] = rounds;
+                    }
+                    else
+                    {
+                        throw new InvalidVoteException(
+                            $"Got vote from unwanted round {vote.Round}",
+                            vote);
+                    }
+                }
+
+                voteSet.AddVerifiedVote(vote);
+            }
+        }
+
+        public VoteSet PreVotes(int round)
+        {
+            lock (_lock)
+            {
+                return GetVoteSet(round, VoteFlag.PreVote);
+            }
+        }
+
+        public VoteSet PreCommits(int round)
+        {
+            lock (_lock)
+            {
+                return GetVoteSet(round, VoteFlag.PreCommit);
+            }
+        }
+
+        // Last round and blockID that has +2/3 prevotes for a particular block or nil.
+        // Returns -1 if no such round exists.
+        public (int, BlockHash) POLInfo()
+        {
+            lock (_lock)
+            {
+                for (int r = _round; r >= 0; r--)
+                {
+                    VoteSet voteSet = GetVoteSet(r, VoteFlag.PreVote);
+                    bool exists = voteSet.TwoThirdsMajority(out BlockHash polBlockHash);
+                    if (exists)
+                    {
+                        return (r, polBlockHash);
+                    }
+                }
+
+                return (-1, default(BlockHash));
+            }
+        }
+
+        /// <summary>
+        /// Gets a <see cref="VoteSet"/> for given round and flag.
+        /// </summary>
+        /// <param name="round">A round of the <see cref="VoteSet"/> to get.</param>
+        /// <param name="voteFlag">A vote flag of the <see cref="VoteSet"/> to get.</param>
+        /// <returns>A <see cref="VoteSet"/> for given round and flag.</returns>
+        /// <exception cref="KeyNotFoundException">Thrown when <see cref="VoteSet"/> for
+        /// given round does not exist.
+        /// </exception>
+        /// <exception cref="ArgumentException">Thrown when given <paramref name="voteFlag"/>
+        /// is not either <see cref="VoteFlag.PreVote"/> or <see cref="VoteFlag.PreCommit"/>.
+        /// </exception>
+        public VoteSet GetVoteSet(int round, VoteFlag voteFlag)
+        {
+            RoundVoteSet roundVoteSet;
+
+            // TODO: Check if try-catch is needed for KeyNotFoundException.
+            roundVoteSet = _roundVoteSets[round];
```

Comment:
Still I think it's better to replace `KeyNotFoundException` with explicit custom exception.

Reply:
IMO its okay, but I'll add `<exception>` XML docs to the method for clearity.